Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -44,76 +51,50 @@ void SoftwareGLContext::gl_begin(GLenum mode)
 {
     APPEND_TO_CALL_LIST_AND_RETURN_IF_NEEDED(gl_begin, mode);
 
-    if (m_in_draw_state) {
-        m_error = GL_INVALID_OPERATION;
-        return;
-    }
-
-    if (mode < GL_TRIANGLES || mode > GL_POLYGON) {
-        m_error = GL_INVALID_ENUM;
-        return;
-    }
+    RETURN_WITH_ERROR_IF(m_in_draw_state, GL_INVALID_OPERATION, );
```

Comment:
The ```, );``` syntax is not very pretty, and it looks like the vast majority of the usages of this macro dont actually have a return value (only 2 instances do), so how about adding a separate RETURN_VALUE_WITH_ERROR_IF or something similar for those ones?

Reply:
Sounds good